atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformer.kt

```diff
@@ -317,11 +386,16 @@ class AtomicFUTransformer(
             super.visitSource(source, debug)
         }
 
-        override fun visitField(access: Int, name: String, desc: String, signature: String?, value: Any?): FieldVisitor? {
+        override fun visitField( access: Int, name: String, desc: String, signature: String?, value: Any?): FieldVisitor? {
             val fieldType = Type.getType(desc)
             if (fieldType.sort == OBJECT && fieldType.internalName in AFU_CLASSES) {
-                val f = fields[FieldId(className, name)]!!
-                val protection = if (f.accessors.isEmpty()) ACC_PRIVATE else 0
+                val fieldId = FieldId(className, name)
+                val f = fields[fieldId]!!
+                val protection = when {
+                    f.hasExternalAccess   -> ACC_PUBLIC or ACC_SYNTHETIC
```

Comment:
Vertical alignment is not used in Kotlin code style. There should be exactly one space before and after `->`